src/ProducerInterface.php

```diff
@@ -14,17 +14,21 @@
 
 namespace Graze\Queue;
 
+use Graze\Queue\Adapter\Exception\UnsupportedOperationException;
 use Graze\Queue\Message\MessageInterface;
 
 interface ProducerInterface
 {
     /**
-     * @return MessageInterface
+     * @param string $body
+     * @param array  $options
      */
     public function create($body, array $options = []);
 
     /**
-     * @param MessageInterface[] $message
+     * @param array $message
      */
     public function send(array $messages);
+
+    public function purge();
```

Comment:
> My thinking here is if we could reasonably assume that all queue implementations have a way of purging then it's fine to put this on ProducerInterface, as a purge operation is similar to a send operation.

So far it's :ok_hand:, but I doubt _every_ queue implementation supports purging.

We're not 1.0.0, so I'm going to go with this and see what Mat thinks if he makes use of it.